spec/graphql/queries/requests/encounters_index_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+module Queries
+  module Encounters
+    RSpec.describe Encounters, type: :request do
+      describe 'encounters' do
+        before(:each) do
+          @list = create_list(:encounter_monster, 6)
+        end
+        it 'returns all encounters associated with a given User ID' do
+          post '/graphql', params: { query: query, variables: { 'userName': "Drizzt" }}
+          json = JSON.parse(response.body)
+          data = json['data']['encounters']
+          expect(data.count).to eq(6)
+          encounter = data.first
+          
+          expect(encounter['userName']).to eq("Drizzt")
+          expect(encounter['encounterName']).to be_a(String)
+          expect(encounter['partySize']).to be_a(Integer)
+          expect(encounter['partyLevel']).to be_a(Integer)
+          expect(encounter['summary']).to be_a(String)
+          expect(encounter['description']).to be_a(String)
+          expect(encounter['treasure']).to be_a(String)
+          expect(encounter['encounterMonsters']).to be_a(Array)
```

Comment:
In your testing, I notice that you used a specific username but just data types for the rest of the elements. Is it normal practice to not actually use specific seed data for testing and check that you get exact matches?

Reply:
This is fairly normal practice, the reason for this is because I'm using a gem called Faker to generate records with randomized data. Most attempts to try and test the actual values will either create a test that won't work, as all the values will change when the test is ran, or it would be testing the encounter keys against our objects attributes (i.e. expect(encounter['encounterName']).to eq(@encounter.encounter_name). The second version can be a good test, but I opted for this method because it shows that the values are not null, and that they are a consistent data type across multiple tests.
I hope this made sense, let me know if you have any follow up questions!

Reply:
Totally made sense, along with the comments @Sykogst mentioned in slack. Thanks for explaining, y'all! ❤️